ios/Classes/NativeBridge/Handlers/ChangePinHandler.swift

```diff
@@ -1,25 +1,30 @@
 import OneginiSDKiOS
 import Flutter
 
-class ChangePinHandler: NSObject {
-    var changePinCompletion: ((Result<Void, FlutterError>) -> Void)?
+class ChangePinHandler {
     private let loginHandler: LoginHandler
     private let registrationHandler: RegistrationHandler
-
-    init(loginHandler: LoginHandler, registrationHandler: RegistrationHandler) {
+    init(_ loginHandler: LoginHandler, _ registrationHandler: RegistrationHandler) {
         self.loginHandler = loginHandler
         self.registrationHandler = registrationHandler
     }
+    func changePin(completion: @escaping (Result<Void, FlutterError>) -> Void) {
+        let delegate = ChangePinDelegateImpl(completion: completion, loginHandler, registrationHandler)
+        SharedUserClient.instance.changePin(delegate: delegate)
+    }
 }
 
-extension ChangePinHandler {
-    func changePin(completion: @escaping (Result<Void, FlutterError>) -> Void) {
-        changePinCompletion = completion
-        SharedUserClient.instance.changePin(delegate: self)
+class ChangePinDelegateImpl: ChangePinDelegate {
+    private let completion: ((Result<Void, FlutterError>) -> Void)
+    private let loginHandler: LoginHandler
+    private let registrationHandler: RegistrationHandler
+
+    init(completion: @escaping (Result<Void, FlutterError>) -> Void, _ loginHandler: LoginHandler, _ registrationHandler: RegistrationHandler) {
```

Comment:
Why did you dropped labels `loginHandler` and `registrationHandler`? BTW my bad, `completion` should be at the end of params list so to me it should looks like
```
init(loginHandler: LoginHandler,
     registrationHandler: RegistrationHandler
     completion: @escaping (Result<Void, FlutterError>) -> Void)